### PR TITLE
Add log_workflow_status_to_cloudwatch in ros2foxy.yml

### PR DESCRIPTION
**Pull request by ahtsan**
*Issue #, if available:*
Now that foxy canaries are in place, we need to add the log_workflow_status_to_cloudwatch to the github action workflow

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/ros2foxy.yml [hunk 1]
```diff
@@ -86,4 +86,54 @@ jobs:
...
       if: ${{ github.event_name == 'schedule' && contains(steps.robot_ws_build.outcome, 'success') && contains(steps.simulation_ws_build.outcome, 'success') }} # Update app-manifest version number if the build was successful
-  
\ No newline at end of file
+  
+
+
+  log_workflow_status_to_cloudwatch:
+  runs-on: ubuntu-latest
+  container:
+    image: ubuntu:bionic
```
Comment (cheng-kevin) [inline .github/workflows/ros2foxy.yml]: should this be ubuntu:focal?

Comment (ahtsan) [reply .github/workflows/ros2foxy.yml]: yes

File: .github/workflows/ros2foxy.yml [hunk 2]
```diff
@@ -86,4 +86,54 @@ jobs:
...
+  needs:
+  - build_and_bundle_ros2
```
Comment (cheng-kevin) [inline .github/workflows/ros2foxy.yml]: the job is named build_and_bundle_ros2_foxy in this workflow

File: .github/workflows/ros2foxy.yml [hunk 3]
```diff
@@ -86,4 +86,54 @@ jobs:
...
+    uses: ros-tooling/action-cloudwatch-metrics@0.0.4
+    with:
+      metric-dimensions: >-
+        [
+          { "Name": "github.event_name", "Value": "${{ github.event_name }}" },
+          { "Name": "github.ref", "Value": "refs/heads/ros2_foxy" },
+          { "Name": "github.repository", "Value": "${{ github.repository }}" }
+        ]
+      namespace: BundleUpload
+      metric-value: ${{ contains(needs.build_and_bundle_ros2.result, 'success') }}
```
Comment (cheng-kevin) [inline .github/workflows/ros2foxy.yml]: build_and_bundle_ros2_foxy

File: .github/workflows/ros2foxy.yml [hunk 4]
```diff
@@ -86,4 +86,54 @@ jobs:
...
+    uses: ros-tooling/action-cloudwatch-metrics@0.0.4
+    with:
+      metric-dimensions: >-
+        [
+          { "Name": "github.event_name", "Value": "${{ github.event_name }}" },
+          { "Name": "github.ref", "Value": "refs/heads/ros2_foxy" },
+          { "Name": "github.repository", "Value": "${{ github.repository }}" }
+        ]
+      namespace: SimulationWorkspaceBuild
+      metric-value: ${{ contains(needs.build_and_bundle_ros2.outputs.simulation_ws_build_result, 'success') }}
```
Comment (cheng-kevin) [inline .github/workflows/ros2foxy.yml]: build_and_bundle_ros2_foxy?

File: .github/workflows/ros2foxy.yml [hunk 5]
```diff
@@ -86,4 +86,54 @@ jobs:
...
+    uses: ros-tooling/action-cloudwatch-metrics@0.0.4
+    with:
+      metric-dimensions: >-
+        [
+          { "Name": "github.event_name", "Value": "${{ github.event_name }}" },
+          { "Name": "github.ref", "Value": "refs/heads/ros2_foxy" },
+          { "Name": "github.repository", "Value": "${{ github.repository }}" }
+        ]
+      namespace: RobotWorkspaceBuild
+      metric-value: ${{ contains(needs.build_and_bundle_ros2.outputs.robot_ws_build_result, 'success') }}
```
Comment (cheng-kevin) [inline .github/workflows/ros2foxy.yml]: build_and_bundle_ros2_foxy?

File: .github/workflows/ros2foxy.yml (hunk 2 above)
Comment (ahtsan) [reply .github/workflows/ros2foxy.yml]: good catch

**Review by cheng-kevin — APPROVED** [review]

**Pull request merged**